packages/edit-site/src/components/save-button/index.js

```diff
@@ -15,32 +20,46 @@ import {
 	currentlyPreviewingTheme,
 	isPreviewingTheme,
 } from '../../utils/is-previewing-theme';
+import { unlock } from '../../lock-unlock';
+
+const { useLocation } = unlock( routerPrivateApis );
+const { useSaveEntities } = unlock( editorPrivateApis );
 
 export default function SaveButton( {
 	className = 'edit-site-save-button__button',
 	variant = 'primary',
 	showTooltip = true,
-	defaultLabel,
+	showReviewMessage,
 	icon,
 	size,
 	__next40pxDefaultSize = false,
 } ) {
-	const { isDirty, isSaving, isSaveViewOpen, previewingThemeName } =
-		useSelect( ( select ) => {
+	const { params } = useLocation();
+	const { setIsSaveViewOpened } = useDispatch( editSiteStore );
+	const {
+		isSaving,
+		isSaveViewOpen,
+		previewingThemeName,
+		dirtyEntityRecords,
+		isOnlyCurrentEntityDirty,
+	} = useSelect(
+		( select ) => {
 			const {
 				__experimentalGetDirtyEntityRecords,
 				isSavingEntityRecord,
 				isResolving,
 			} = select( coreStore );
-			const dirtyEntityRecords = __experimentalGetDirtyEntityRecords();
+			const { hasNonPostEntityChanges, isEditedPostDirty } =
+				select( editorStore );
+			const _dirtyEntityRecords = __experimentalGetDirtyEntityRecords();
 			const { isSaveViewOpened } = select( editSiteStore );
 			const isActivatingTheme = isResolving( 'activateTheme' );
 			const currentlyPreviewingThemeId = currentlyPreviewingTheme();
-
 			return {
-				isDirty: dirtyEntityRecords.length > 0,
+				dirtyEntityRecords: _dirtyEntityRecords,
+				hasDirtyEntities: !! _dirtyEntityRecords.length,
```

Comment:
We could return `dirtyEntitiesCount` or something similar here. Also `hasDirtyEntities` is returned by `mapSelect` but never actually used.

Reply:
Good catch! This was a leftover from all the moving around. I used `dirtyEntitiesCount`.